src/nvim/lua/vim.lua

```diff
@@ -279,7 +279,19 @@ setmetatable(vim, {
 
 do
   local a = vim.api
+  local nvim_command = a.nvim_command
   local validate = vim.validate
+  vim.cmd = setmetatable({}, {
+    __index = function(_, k)
+      return function(s)
+        validate { cmd = {s, 's', true} }
+        if s == nil then
+          return nvim_command(k)
+        end
+        return nvim_command(table.concat({k, s}, ' '))
```

Comment:
Isn't this just `(k..' '..s)` ?

Reply:
table.concat is faster since it can calculate the final size ahead of time and requires one allocation. repeated `..` requires growing a buffer twice.

not that it matters much here, but it's an automatic thing for me to do.

Reply:
Is this really the case here? `table.concat` requires allocating an extra table (in this case). It would seem very odd that luajit would micro-optimize the syntax of contact+table-literal but not two explicit `..` (the case of a loop is of course different, but that is not what we have here).

Reply:
I checked the IR. You're right. I think table.concat is faster in some circumstances, but this isn't one of them.